pkg/apis/serving/v1alpha1/kfservice_status.go

```diff
@@ -87,6 +87,11 @@ func (ss *KFServiceStatus) PropagateCanaryConfigurationStatus(canaryConfiguratio
 	}
 }
 
+func (ss *KFServiceStatus) ResetCanaryConfigurationStatus() {
+	ss.Canary = StatusConfigurationSpec{}
+	conditionSet.Manage(ss).MarkUnknown(CanaryPredictorReady, "CanarySpecUnavailable", "Canary spec is missing")
```

Comment:
Does this save a status condition that says "unknown" in the CRD if there is no canary? I'd expect to instead clear the status condition altogether. I.e., it shouldn't say anything if canary isn't specified.

Reply:
I am not sure why there is no RemoveCondition in https://github.com/knative/pkg/blob/master/apis/condition_set.go#L50

Reply:
@yuzisun

Reply:
I am chatting with Matt from knative team

Reply:
mattmoor [3:36 PM]
You can use `{Get,Set}Conditions()` methods, but we don't typically elide conditions like that
Generally all of the "terminal" conditions are present

Reply:
Do we need to model this as a terminatable condition? I.e., CanaryUnspecified? This seems somewhat odd.

Reply:
Dan was suggesting to add "RemoveCondition" to knative/pkg upstream. Then update pkg and use it here.

Reply:
SGTM

Reply:
`Canary spec is missing` sounds like an error, maybe `Canary spec unavailable` ?